src/backend/cdb/cdbllize.c

```diff
@@ -590,8 +590,26 @@ ParallelizeCorrelatedSubPlanMutator(Node *node, ParallelizeCorrelatedPlanWalkerC
 				scanPlan->flow->flotype = FLOW_SINGLETON;
 			}
 
-			broadcastPlan(scanPlan, false /* stable */ , false /* rescannable */,
-						  ctx->currentPlanFlow->numsegments /* numsegments */);
+			/*
+			 * Replicated locus can't be reduced to SingleQE as it's done for
+			 * SegmentGeneral locus, because it's executed at full n-gang
+			 * Therefore, in case of volatile quals we must focus Replicated
+			 * plan first, apply the quals, and then broadcast it back.
+			 */
+			if (scanPlan->flow->locustype == CdbLocusType_Replicated &&
```

Comment:
During the private talk with @Stolb27 , I figured out that promising solution was to throw an error here with possible creation of explicit broadcast in the future.
Current solution works, but looks like a good old crutch. Why you switched?